internal/db/arango.go

```diff
@@ -118,14 +140,96 @@ func (c *ArangoClient) QueryAnnotations(ctx context.Context, key string) ([]docu
 
 	var annotations []documents.Annotation
 	for {
-		var doc documents.Annotation
-		_, err := cursor.ReadDocument(ctx, &doc)
+		_, err := cursor.ReadDocument(ctx, &annotations)
 		if driver.IsNoMoreDocuments(err) {
 			break
 		} else if err != nil {
 			return nil, err
 		}
-		annotations = append(annotations, doc)
 	}
+
 	return annotations, nil
 }
+
+func (c *ArangoClient) QueryScoreByLayer(ctx context.Context, key string, layer contracts.LayerType) ([]documents.Score, error) {
+	db, err := c.instance.Database(ctx, c.cfg.DatabaseName)
+	if err != nil {
+		return nil, err
+	}
+
+	var query string
+	switch layer {
+	case contracts.Application:
+		query = `FOR s IN scores FILTER s.dataRef == @key AND s.layer == @layer RETURN [s]`
+	case contracts.CiCd:
+		query = `FOR appScore IN scores FILTER appScore.dataRef == @key 
+				LET cicdScore = (
+					FOR s IN scores FILTER 
+					s.layer == @layer AND s.tag ANY IN appScore.tag 
+					RETURN s 
+				)
+				RETURN cicdScore `
+	case contracts.Os, contracts.Host:
+		query = `FOR a in annotations FILTER a.dataRef == @key LIMIT 1
+				LET scores = (FOR s IN scores FILTER s.layer == @layer AND a.host IN s.tag RETURN s)
```

Comment:
I believe the same thing can happen with Host and Os scores
```suggestion
				LET scores = (FOR s IN scores FILTER s.layer == @layer AND a.host IN s.tag SORT s.timestamp DESC LIMIT 1 RETURN s)

```

Reply:
I'd have the same comment as above.